cpp/oneapi/dal/backend/interop/data_conversion.cpp

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/backend/interop/data_conversion.hpp"
+#include "oneapi/dal/exceptions.hpp"
+
+namespace daal_dm = daal::data_management;
+
+namespace oneapi::dal::backend::interop {
+
+daal_dm::features::IndexNumType get_daal_index_num_type(data_type t) {
+    switch (t) {
+        case data_type::int32: return daal_dm::features::DAAL_INT32_S;
+        case data_type::int64: return daal_dm::features::DAAL_INT64_S;
+        case data_type::uint32: return daal_dm::features::DAAL_INT32_U;
+        case data_type::uint64: return daal_dm::features::DAAL_INT64_U;
+        case data_type::float32: return daal_dm::features::DAAL_FLOAT32;
+        case data_type::float64: return daal_dm::features::DAAL_FLOAT64;
+        default: return daal_dm::features::DAAL_OTHER_T;
+    }
+}
+
+daal_dm::internal::ConversionDataType get_daal_conversion_data_type(data_type t) {
+    switch (t) {
+        case data_type::int32: return daal_dm::internal::DAAL_INT32;
+        case data_type::float32: return daal_dm::internal::DAAL_SINGLE;
+        case data_type::float64: return daal_dm::internal::DAAL_DOUBLE;
+        default: return daal_dm::internal::DAAL_OTHER;
+    }
+}
+
```

Comment:
```suggestion
inline void is_converatble(data_type src_type, data_type dst_type) {
    if (src_type == daal_dm::features::DAAL_OTHER_T ||
        dst_type == daal_dm::internal::DAAL_OTHER_T) {
            throw internal_error("unsupported conversion types");
    }
}
```

Reply:
1. You have a misprint in the name of this function
2. enums in `daal_dm::features` and `daal_dm::internal` are **different** - there is no value of `daal_dm::internal::DAAL_OTHER_T`
3. I do not see why I need to extract this check to the separate function - it is used only once. Method extraction increases code complexity in this case